tests/Functional/Controller/StaticRoutesSmokeTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\Functional\Controller;
+
+use PHPUnit\Framework\Attributes\DataProvider;
+use Symfony\Bundle\FrameworkBundle\Test\TestContainer;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Extracted and simplified from the "pierstoval/smoke-testing" package.
+ *
+ * @see https://github.com/Pierstoval/SmokeTesting
+ *
+ * @author Alex "Pierstoval" Rock <pierstoval@gmail.com>
+ */
+class StaticRoutesSmokeTest extends WebTestCase
+{
+    #[DataProvider('provideRouteCollection')]
+    public function testRoutesDoNotReturnInternalError(string $httpMethod, string $routeName, string $routePath): void
+    {
+        $client = static::createClient();
+
+        $client->request($httpMethod, $routePath);
+
+        $response = $client->getResponse();
+        static::assertLessThan(
+            500,
+            $response->getStatusCode(),
+            \sprintf('Request "%s %s" for route "%s" returned an internal error.', $httpMethod, $routePath, $routeName),
+        );
+    }
+
+    /**
+     * @return \Generator<string, Route>
+     */
+    public static function provideRouteCollection(): \Generator
+    {
+        if (!is_a(self::class, WebTestCase::class, true)) {
+            throw new \RuntimeException(\sprintf('The "%s" trait trait can only be used in an instance of "%s"', self::class, WebTestCase::class));
+        }
+
+        static::bootKernel();
+
+        /** @var TestContainer $container */
```

Comment:
Not sure these @var annotations are needed with the Symfony PHPStan plugin, to test.

Reply:
It's indeed not necessary for PHPStan, but it might be necessary for some IDEs that don't understand Symfony's DI rules. Some static analysers relying only on PHPDoc will also understand annotations better. I think we should keep it.